packages/jellyfish-api-core/__tests__/category/net/getPeerInfo.test.ts

```diff
@@ -0,0 +1,92 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { PeerInfo } from 'packages/jellyfish-api-core/src/category/net'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+createNetworkTests('Network without masternode', new RegTestContainer())
+createNetworkTests('Network on masternode', new MasterNodeRegTestContainer())
+
+function createNetworkTests (name: string, container: RegTestContainer | MasterNodeRegTestContainer): void {
+  describe(name, () => {
+    const client = new ContainerAdapterClient(container)
+
+    beforeAll(async () => {
+      await container.start()
+      await container.addNode(await container.getIp('bridge'))
```

Comment:
hmm.. here should be like this
await container.addNode(await container.getIp('testcontainer-{num}')

Reply:
@canonbrother thanks for the feedback, I have some questions on this

```ts
await container.addNode(await container.getIp('testcontainer-{num}')
```

When I looked at [DockerContainer](https://github.com/DeFiCh/jellyfish/blob/main/packages/testcontainers/src/containers/DockerContainer.ts#L18-L22) it seems to require a name that exists in the networks list. When I printed out the network lists value in this test file the only name that exists is `bridge`.

I did see that in [ContainerGroup.add](https://github.com/DeFiCh/jellyfish/blob/main/packages/testcontainers/src/containers/RegTestContainer/ContainerGroup.ts#L75) its using name assigned when creating the docker network at [ContainerGroup.start](https://github.com/DeFiCh/jellyfish/blob/main/packages/testcontainers/src/containers/RegTestContainer/ContainerGroup.ts#L29)

So when executing `start` directly on [MasterNode](https://github.com/DeFiCh/jellyfish/blob/main/packages/testcontainers/src/containers/RegTestContainer/Masternode.ts) or [RegTestContainer](https://github.com/DeFiCh/jellyfish/blob/main/packages/testcontainers/src/containers/RegTestContainer/index.ts) instances I do not see any network assignment options as part of it.

So I am wondering if perhaps I should be using the ContainerGroup in this test perhaps?

Reply:
we don't ContainerGroup as the containers are synced once they're started
we need containers which are not synced to test `addnode`

maybe out of the box.. i added the docker network op
refer to https://github.com/DeFiCh/jellyfish/pull/938/files#r779342382

Reply:
Markings as resolved following updates from https://github.com/DeFiCh/jellyfish/pull/938#discussion_r779342382